clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/mapreduce/datawriter/jobs/DataWriterMRJob.java

```diff
@@ -0,0 +1,394 @@
+    } else {
+      conf.setInt(VALUE_SCHEMA_ID_PROP, pushJobSetting.valueSchemaId);
+      conf.setInt(DERIVED_SCHEMA_ID_PROP, pushJobSetting.derivedSchemaId);
+    }
+    conf.setBoolean(ENABLE_WRITE_COMPUTE, pushJobSetting.enableWriteCompute);
+
+    if (!props.containsKey(KAFKA_PRODUCER_REQUEST_TIMEOUT_MS)) {
+      // If the push job plug-in doesn't specify the request timeout config, default will be infinite
+      conf.set(KAFKA_PRODUCER_REQUEST_TIMEOUT_MS, Integer.toString(Integer.MAX_VALUE));
+    }
+    if (!props.containsKey(KAFKA_PRODUCER_RETRIES_CONFIG)) {
+      // If the push job plug-in doesn't specify the retries config, default will be infinite
+      conf.set(KAFKA_PRODUCER_RETRIES_CONFIG, Integer.toString(Integer.MAX_VALUE));
+    }
+    if (!props.containsKey(KAFKA_PRODUCER_DELIVERY_TIMEOUT_MS)) {
+      // If the push job plug-in doesn't specify the delivery timeout config, default will be infinite
+      conf.set(KAFKA_PRODUCER_DELIVERY_TIMEOUT_MS, Integer.toString(Integer.MAX_VALUE));
+    }
+
+    conf.set(TELEMETRY_MESSAGE_INTERVAL, props.getString(TELEMETRY_MESSAGE_INTERVAL, "10000"));
+    conf.setBoolean(EXTENDED_SCHEMA_VALIDITY_CHECK_ENABLED, pushJobSetting.extendedSchemaValidityCheckEnabled);
+
+    // Compression related
+    // Note that COMPRESSION_STRATEGY is from topic creation response as it might be different from the store config
+    // (eg: for inc push)
+    conf.set(
+        COMPRESSION_STRATEGY,
+        pushJobSetting.topicCompressionStrategy != null
+            ? pushJobSetting.topicCompressionStrategy.toString()
+            : CompressionStrategy.NO_OP.toString());
+    conf.set(
+        ZSTD_COMPRESSION_LEVEL,
+        props.getString(ZSTD_COMPRESSION_LEVEL, String.valueOf(Zstd.maxCompressionLevel())));
+    conf.setBoolean(ZSTD_DICTIONARY_CREATION_SUCCESS, pushJobSetting.isZstdDictCreationSuccess);
+
+    /** Allow overriding properties if their names start with {@link HADOOP_PREFIX}.
+     *  Allow overriding properties if their names start with {@link VeniceWriter.VENICE_WRITER_CONFIG_PREFIX}
+     *  Allow overriding properties if their names start with {@link ApacheKafkaProducerConfig.KAFKA_CONFIG_PREFIX}
+     *  Allow overriding properties if their names start with {@link KafkaInputRecordReader.KIF_RECORD_READER_KAFKA_CONFIG_PREFIX}
+     **/
+    List<String> passThroughPrefixList = Arrays.asList(
+        VeniceWriter.VENICE_WRITER_CONFIG_PREFIX,
+        KAFKA_CONFIG_PREFIX,
+        KafkaInputRecordReader.KIF_RECORD_READER_KAFKA_CONFIG_PREFIX);
+    int passThroughPrefixListSize = passThroughPrefixList.size();
+    if (passThroughPrefixListSize > 1) {
+      /**
+       * The following logic will make sure there are no two prefixes, which will be fully overlapped.
+       * The algo is slightly inefficient, but it should be good enough for VPJ.
+       */
+      for (int i = 0; i < passThroughPrefixListSize; ++i) {
+        for (int j = 0; j < passThroughPrefixListSize; ++j) {
```

Comment:
nit: j can start from i + 1, and below we can do two startsWith check. It is slightly better.

Reply:
But actually I don't understand the check and the comment, could you give an example?

Reply:
This code has been copied over from `VenicePushJob` class so I didn't look through it in detail: https://github.com/linkedin/venice/blob/d788799205965e267cdf278a418f21c9375e7fad/clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java#L3090-L3106

It looks like it checks if no `passThroughPrefix` is a prefix of any other `passThroughPrefix`. So, we want to check `prefix1.prefix2` is not allowed if `prefix1` is in the list. Similarly, if `prefix2` is also on the list, then we should not allow `prefix1.prefix2` or `prefix2.prefix1`, so, we can't do `j = i+1; ...` type of loop

Reply:
I re-read your comment and I'll make that change

Reply:
I also feel  that this check is not necessary, as the prefixes are constants. But I didn't make any functional change to this part in the refactoring. I feel we could later move these checks to a test case instead of running it in a running VPJ job